webpush/encrypt.go

```diff
@@ -270,7 +281,7 @@ func hkdf(salt, ikm, info []byte, length int) []byte {
 	mac = hmac.New(sha256.New, prk)
 	mac.Write(info)
 	mac.Write([]byte{1})
-	return mac.Sum(nil)[0:length]
+	return mac.Sum(nil)[0:length], nil
```

Comment:
I'd say this is the only place where you want to verify length is not > than `len(mac.Sum(nil))` but only to avoid panic.